elements/src/main/java/com/manydesigns/elements/crypto/KeyManager.java

```diff
@@ -80,22 +80,29 @@ private KeyManager(Configuration configuration) throws GeneralSecurityException,
     }
 
     private void checkOrSaveHash(String path, String key) throws IOException, GeneralSecurityException, InvalidPassphraseException {
-        File checksumFile = new File(path + ".sum");
-
         MessageDigest digest = MessageDigest.getInstance("SHA-256");
-        digest.update(path.getBytes(StandardCharsets.UTF_8));
-        String checksum = CryptoUtils.getStringChecksum(digest, key);
 
-        if (checksumFile.exists()) {
-            String current = StringUtils.trimToEmpty(CryptoUtils.getKey(checksumFile.getAbsolutePath()));
+        String checksum = System.getenv("PORTOFINO_PASSPHRASE_SUM");
+        if (checksum == null) {
+            digest.update(path.getBytes(StandardCharsets.UTF_8));
```

Comment:
Non capisco la logica qui. Mi pare che il successivo getStringChecksum, chiamando digest.digest, renda ininfluente l'update. Ma magari mi sfugge qualcosa

Reply:
facendo delle prove ho notato che nel calcolare il checksum della chiave, veniva preso in considerazione (giustamente) anche il path dello stesso.
Con un po di debug ho capito che era questa update del digest.

Reply:
OK